File: framework/ecommerce-framework-api/src/main/java/com/sdl/ecommerce/api/model/Location.java
package com.sdl.ecommerce.api.model;

import java.util.List;

/**
 * Location
 *
 * @author nic
 */
public interface Location {

    /**
     * Get category reference
     * @return reference
     */
    CategoryRef getCategoryRef();

    /**
     * Get facet parameters
     * @return list of facet parameters
     */
    // TODO: Should this be Facet interface instead
    List<FacetParameter> getFacets();

    /**
     * Get product reference
     * @return reference
     */
    ProductRef getProductRef();

    /**
     * Static URL managed from the E-Commerce system
     * @return
     */
    String getStaticUrl();
}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.OData/Model/Location.cs
using SDL.ECommerce.Api.Model;

using System.Collections.Generic;
using System.Linq;

using SDL.ECommerce.Api;

namespace SDL.ECommerce.OData
{
    public partial class Location : ILocation
    {
        private IList<FacetParameter> facets;
          
        ICategoryRef ILocation.CategoryRef
        {
            get
            {
                return this.CategoryRef;
            }
        }

        IList<FacetParameter> ILocation.Facets
        {
            get
            {
                if ( this.Facets != null && facets == null )
                {
                    facets = this.Facets.Select(facet => FacetParameter.FromUrl(facet)).ToList();
                }
                return facets;
            }
        }

        IProductRef ILocation.ProductRef
        {
            get 
            {
                return this.ProductRef;
            }
        }
    
    }
}
